htdocs/core/lib/security.lib.php

```diff
@@ -331,10 +331,17 @@ function restrictedArea($user, $features, $objectid=0, $tableandshare='', $featu
     // is linked to a company allowed to $user.
     if (! empty($objectid) && $objectid > 0)
     {
-    	$ok = checkUserAccessToObject($user, $featuresarray, $objectid, $tableandshare, $feature2, $dbt_keyfield, $dbt_select);
-		return $ok ? 1 : accessforbidden();
+    	if (!checkUserAccessToObject($user, $featuresarray, $objectid, $tableandshare, $feature2, $dbt_keyfield, $dbt_select))
+			accessforbidden();
     }
 
+	// get more permissions checks from hooks
+	global $hookmanager;
+	$hookmanager->initHooks(array('permissions'));
+	$parameters=array('features'=>$features,'objectid'=>preg_replace("/'/", '', $objectid),'idtype'=>$dbt_select);
+	$reshook=$hookmanager->executeHooks('restricted',$parameters);
+	if ($reshook < 0) accessforbidden();
+
```

Comment:
The standard of hooks is to introduce only "adrelace" hooks that return a non zero value to say replace standard code.
So you must first call the hook and then if reshook is 0 you return the result of standard code checkuseraccestoobject.

Reply:
The hook is to replace or complete method  "checkUserAccessToObject"
So it must be called before calling checkUserAccessToObject. And the if return of ok is 0, check must also be done by calling checkUserAccessToObject
Returning if permission is access is ok or not can be done into the resarray